metals/src/main/scala/scala/meta/internal/metals/BuildTargets.scala

```diff
@@ -202,6 +202,15 @@ final class BuildTargets() {
   ): Option[AbsolutePath] =
     buildServerOf(buildTarget).map(_.workspaceDirectory)
 
+  def canCompile(id: BuildTargetIdentifier): Boolean = {
+    targetData(id).exists { data =>
```

Comment:
@ckipp01 `data.iterable.exists` would return the first build data that was true and since there is an empty AmmoniteBuildData that would alywas be true. We need to first get the data for a specific target id.

Reply:
O duh! Of course! 😅

Reply:
It's always an issue like this with `exists` and `forall` :sweat_smile: It's easier to figure out from the backseat